pkg/reconciler/brokercell/broker_targets_config.go

```diff
@@ -133,6 +136,36 @@ func (r *Reconciler) addToConfig(ctx context.Context, b *brokerv1beta1.Broker, t
 	})
 }
 
+func targetsConfigMapEqual(cm1, cm2 *corev1.ConfigMap) bool {
+	if !equality.Semantic.DeepEqual(cm1.Data, cm2.Data) {
+		return false
+	}
+	// The broker targets ConfigMap BinaryData holds the serialized TargetsConfig
+	// proto, and therefore cannot be safely compared with equality.Semantic.DeepEqual.
+	// Instead, use proto.Equal to compare protos.
+	if len(cm1.BinaryData) != len(cm2.BinaryData) {
+		return false
+	}
```

Comment:
Should we skip this? If binary aren't equal, it doesn't necessarily mean proto aren't equal, right?

Reply:
The length comparison just checks that the number of keys in the BinaryData maps are the same, it doesn't compare the binary data itself.

I guess since this function is specialized for the broker-targets ConfigMap, the function can be written more specifically to only consider the appropriate key. How does that sound?

If you mean the Data deep equal, I can remove it, since it is redundant.

Reply:
Both points sound good to me.

Reply:
Done.